server/StudySharp.ApplicationServices/ValidationRules/TagRules.cs

```diff
@@ -14,9 +15,14 @@ public TagRules(StudySharpDbContext context)
             _context = context;
         }
 
-        public async Task<bool> IsNameUniqueAsync(string name)
+        public async Task<bool> IsNameUniqueAsync(string name, CancellationToken cancellationToken)
         {
-            return !await _context.Tags.AnyAsync(_ => _.Name == name);
+            return !await _context.Tags.AnyAsync(_ => _.Name == name, cancellationToken);
+        }
+
+        public async Task<bool> IsTagIdExistAsync(int id, CancellationToken cancellationToken)
+        {
+            return await _context.Tags.FindAsync(id, cancellationToken) != null;
```

Comment:
Prefer using AnyAsync in these cases because FindAsync retrieves entity from the database and AnyAsync just identifies if an entity exists